### PR TITLE
Resolving Concurrency Issues 

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Resolution for these github issues: https://github.com/Azure/azure-storage-blob-go/issues/260, https://github.com/Azure/azure-storage-blob-go/issues/257

Will add additional tests and comments!

### CODE REVIEW

File: azblob/zt_highlevel_test.go [hunk 1]
```diff
@@ -432,3 +432,31 @@ func (s *aztestsSuite) TestDoBatchTransferWithError(c *chk.C) {
 	mmf.isClosed = true
 	time.Sleep(time.Second * 5)
 }
+
+func (s *aztestsSuite) Test_CopyFromReader(c *chk.C) {
+	ctx := context.Background()
+	p, err := createSrcFile(_1MiB * 12)
+	if err != nil {
+		panic(err)
```
Comment (zezha-msft) [inline azblob/zt_highlevel_test.go:440]: Assert instead of panic.

**Review by zezha-msft — APPROVED** [review]

File: azblob/zt_highlevel_test.go [hunk 2]
```diff
@@ -432,3 +432,31 @@ func (s *aztestsSuite) TestDoBatchTransferWithError(c *chk.C) {
...
+	}
+
+	br := newFakeBlockWriter()
+	defer br.cleanup()
+
+	br.errOnBlock = 1
+	transferManager, err := NewStaticBuffer(_1MiB, 1)
+	if err != nil {
+		panic(err)
```
Comment (mohsha-msft) [inline azblob/zt_highlevel_test.go:456]: c.assert(err, chk.NotNil)

File: azblob/zt_highlevel_test.go [hunk 3]
```diff
@@ -432,3 +432,31 @@ func (s *aztestsSuite) TestDoBatchTransferWithError(c *chk.C) {
 	mmf.isClosed = true
 	time.Sleep(time.Second * 5)
 }
+
+func (s *aztestsSuite) Test_CopyFromReader(c *chk.C) {
+	ctx := context.Background()
+	p, err := createSrcFile(_1MiB * 12)
```
Comment (mohsha-msft) [inline azblob/zt_highlevel_test.go:438]: Just Curious: Why `12MiB`? If this any random number, can we keep the size a little less as it increases testsuite runtime on local

**Review by mohsha-msft — APPROVED** [review]

File: azblob/zt_highlevel_test.go (hunk 3 above)
Comment (siminsavani-msft) [reply azblob/zt_highlevel_test.go:438]: So I was hoping to test a larger file, but we could potentially keep it smaller.

**Review by adreed-msft — APPROVED** [review]

**Pull request merged**